libs/core/src/ScaleFactorsManager.cpp

```diff
@@ -262,7 +262,7 @@ float ScaleFactorsManager::TryToEvaluate(const CorrectionRef &correction, const
 }
 
 float ScaleFactorsManager::GetPileupScaleFactorCustom(int nVertices) {
-  if (!applyScaleFactors["pileup"][0]) return 1.0;
+  if (!ShouldApplyScaleFactor("PUjetID")) return 1.0;
```

Comment:
Is this correct? The method name mentions "pileup", but you check for "PUjetID", which is not the same.